content/academy/apify_platform/deploying/inputs_outputs.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Inputs & outputs
+description: Learn to accept input into your actor, do something with it, then return output. Actors can be written in any language, so this concept is language agnostic.
+menuWeight: 1
+paths:
+    - apify-platform/deploying/inputs-outputs
+---
+
+# [](#inputs-outputs) Inputs & outputs
+
+Most of the time when you're creating a project, you are expecting some sort of input off of which your software will run off of. Often times as well, you want to provide some sort of output once your software has completed running. With Apify, it is extremely easy to take in inputs and deliver outputs.
+
+An important thing to understand regarding inputs and outputs is that they are read/written differently depending on where the actor is running:
+
+- If your actor is running locally, the inputs/outputs are usually provided in the filesystem, and environment variables are injected either by you, the developer, and by the Apify CLI by running the project with the `apify run` command.
+
+- While running in a Docker container on the platform, environment variables are automatically injected, and inputs & outputs are provided and modified using Apify's REST API.
+
+## [](#accepting-input) Accepting input
+
+There are multiple ways to accept input into your project. The option you go with depends on the language you have written your project in. If you are using Node.js for your repo's code, you can use the [Apify SDK](https://www.npmjs.com/package/apify). Otherwise, you can use the useful environment variables automatically set up for you by Apify to write utility functions which read the actor's input and return it.
+
+### Accepting input with the Apify SDK
+
+Since we're using Node.js, let's install the **apify** package by running the following command:
+
+```shell
+npm install apify
+```
+
+Now, let's import `Apify` and use the `Apify.getInput()` function to grab our input.
+
+```JavaScript
+// index.js
+import Apify from 'apify';
+
+const input = await Apify.getInput();
+console.log(input);
+```
+
+If we run this right now, we'll see **null** in our terminal - this is because we never provided any sort of test input, which should be provided in the default key-value store. The `Apify.getInput()` function has detected that there is no **apify_storage** folder and generated one for us. We'll now add an **INPUT.json** file within **apify_storage/key_value_stores/default** to match what we're expecting in our code.
+
+```JSON
+{
+    "numbers": [5, 5, 5, 5]
+}
+```
+
+Then we can modify our code to grab the input and use it to generate a solution which is logged to the console.
```

Comment:
```suggestion
Then we can add our example project code from earlier. It will grab the input and use it to generate a solution which is logged to the console.
```